ci(shared-data, components): add github action for publishing packages to npm

b-cooper opened the pull request:
# Overview

In order to support a single source of truth for applications outside of the monorepo (read:
"Protocol Library") that rely on fundamental components and geometries within the mono repo, add tag
based deploy actions that publish bundled artifacts to NPM.

# Changelog


# Review requests

# Risk assessment

b-cooper committed:
components lib publish workflow starter

b-cooper committed:
remove env var setting

b-cooper committed:
attempt to swap out version in package.json

b-cooper committed:
yaml indent fix

b-cooper committed:
store version as env var

b-cooper committed:
remove unnecessary braces

b-cooper committed:
remove setup step

b-cooper committed:
remove explicit npm install

b-cooper committed:
install json package

b-cooper committed:
use sed to trim

sfoster1 commented on .github/workflows/components-test-build-deploy.yaml:
```diff
@@ -137,3 +158,41 @@ jobs:
+    needs: ['js-unit-test', 'determine-build-type']
+    if: needs.determine-build-type.outputs.type == 'publish'
+    env: 
+      VERSION_STRING: echo github.ref | sed 's/refs\/tags\/components@//'
```
I don't think you can drop shell commands here, I think it's [expressions](https://docs.github.com/en/actions/learn-github-actions/expressions#functions) only. i'd recommend adding the tag value as an output from the `determine-build-type` job, where you can use shell

sfoster1 commented on .github/workflows/components-test-build-deploy.yaml:
```diff
@@ -137,3 +158,41 @@ jobs:
+        run: |
+          npm install -g json
+          json -I -f ./components/package.json -e "this.version=\"$VERSION_STRING\""
+      - run: cd ./components && npm publish --provenace --access public
```
```suggestion
      - run: cd ./components && npm publish --provenance --access public
```

b-cooper committed:
inline bash var

b-cooper committed:
name deploy step

b-cooper committed:
explicitly set registry url

b-cooper committed:
add shared data publish task

b-cooper committed:
update publish if statement

b-cooper committed:
set github output correctly and add echo

b-cooper committed:
link up components to shared data deps

b-cooper committed:
else if => elif

b-cooper committed:
shared data parse both trigger tags

b-cooper committed:
regex match in shared data publish action

b-cooper committed:
add build step for shared data as lib

b-cooper committed:
make lib for components and shared-data

b-cooper committed:
only yarn no setup-js

b-cooper committed:
npm publish instead

b-cooper committed:
tweak some stuff

b-cooper committed:
Merge branch 'edge' into chore_add-npm-deploy-sd-and-cl

b-cooper committed:
attempt to triangulate npm pub failure

b-cooper committed:
add back in redundant yarn cache step

b-cooper committed:
change order of pub calls

b-cooper committed:
inline token injection

b-cooper committed:
back to yarn

b-cooper committed:
minimal updates from working version test

b-cooper committed:
fix yaml error

b-cooper committed:
yarn install for publish step

b-cooper committed:
move env up

b-cooper committed:
create yarnrc locally

b-cooper committed:
proof of fix

b-cooper committed:
npm pub

b-cooper committed:
cp npmrc to local

b-cooper committed:
make npmrc anew

b-cooper committed:
remove merge conflict

b-cooper committed:
setup node pre components publish

b-cooper committed:
filename to shared data webpack config

b-cooper committed:
exclude ts source from shared data lib

b-cooper commented on Makefile:
```diff
@@ -61,7 +61,6 @@ setup-js:
 	yarn
 	$(MAKE) -C $(APP_SHELL_DIR) setup
 	$(MAKE) -C $(APP_SHELL_ODD_DIR) setup
-	$(MAKE) -C $(SHARED_DATA_DIR) setup-js
```
this call is no longer necessary.

b-cooper commented on shared-data/Makefile:
```diff
@@ -28,17 +25,12 @@ clean: clean-js clean-py
+lib-js:
+	yarn webpack
 
-.PHONY: dist-js
```
these were purely used for bundling v1 labware defs for old versions of monorepo SPA's

b-cooper committed:
ci(shared-data, components): add github action for publishing packages to npm

In order to support a single source of truth for applications outside of the monorepo (read:
"Protocol Library") that rely on fundamental components and geometries within the mono repo, add tag
based deploy actions that publish bundled artifacts to NPM.

b-cooper committed:
remove old v1 def affordances

b-cooper commented on protocol-library-kludge/src/URLDeck.tsx:
```diff
@@ -115,14 +106,7 @@ export class URLDeck extends React.Component<{}> {
                   >
                     {labwareDefV2 ? (
                       <LabwareRender definition={labwareDefV2} />
-                    ) : (
```
this whole directory will never be deployed again. It can be removed when legacy PL is unlisted

sfoster1 replied:
We eagerly await the day

sfoster1 reviewed (approved):
Looks good to me!

b-cooper committed:
remove unused import

b-cooper committed:
format and import fix

b-cooper committed:
fix up test mock

shlokamin commented on .github/workflows/components-test-build-deploy.yaml:
```diff
@@ -102,11 +102,32 @@ jobs:
+              echo 'type=none' >> $GITHUB_OUTPUT
+            fi
+
   deploy-components:
```
```suggestion
  deploy-components-storybook:
```

shlokamin reviewed (approved):
🥳
